examples/resources/kentik-synthetic_test-bulk-create/agents_filtered_by_asn/resource.tf

```diff
@@ -0,0 +1,56 @@
+// Create a test with list of agents that have given ASN.
+// Inputs: list of asns
+
+data "kentik-synthetics_agents" "agents" {}
+
+locals {
+  asnList = [20473, 11111, 12333]
+
+  agents_ids = compact([for agent in data.kentik-synthetics_agents.agents.items:
+                          contains(local.asnList, agent.asn) ? agent.id : ""])
+}
+
+resource "kentik-synthetics_test" "private-agents-test" {
+  name      = "agents-filtered-by-asn-test"
+  type      = "hostname"
+  device_id = "75702"
```

Comment:
The API (unfortunately) requires the `device_id` attribute in test creation methods, but it actually ignores it (accepts any value). Can we make the provider default `device_id` to 0 if it is not explicitly provided? Once the default is in place, we should remove `device_id` from all examples. Ideally, we should completely hide the attribute, because even the value returned by the API is of no use to anybody. It is really just a relic required by the back-end which the API unfortunately leaks :-/.

Reply:
If I set default `device_id` to 0 like so:

Reply:
I already responded on Slack DM, but how about removing device_id from schema and setting device_id to "0" while constructing test object during create? For update, we could remove device_id from field mask.

Reply:
That is what I had in mind. Entirely hide the attribute from Terraform, so that it is never shown to users and there is no way to set it in a resource file. Always set it to 0 in any request that requires it. Do we really need to do anything with field mask on update, if we never allow users to modify the attribute?

Reply:
I would remove device_id from field mask to avoid unintended consequences, such as unsetting/zeroing the device_id in the server. It depends on server's implementation.

Reply:
Ok, it seems to me that this issue deserves its own Jira task. https://clkentik.atlassian.net/browse/KNTK-299